sunpy/timeseries/sources/goes.py

```diff
@@ -214,14 +182,53 @@ def _parse_hdus(cls, hdulist):
                              ('xrsb', u.W/u.m**2)])
         return data, header, units
 
+    @staticmethod
+    def _parse_netcdf(filepath):
+        """
+        Parses the netCDF GOES files to return the data, header and associated units.
+
+        Parameters
+        ----------
+        filepath : `~str`
+            The path of the file to parse
+        """
+        import h5netcdf
+
+        d = h5netcdf.File(filepath, mode="r")
+
+        header =  MetaDict(OrderedDict(d.attrs))
+        if "a_flux" in d.variables:
+            xrsa = np.array(d["a_flux"])
+            xrsb = np.array(d["b_flux"])
+            times = parse_time("1970-01-01") + TimeDelta(d["time"], format="sec")
+        elif "xrsa_flux" in d.variables:
+            xrsa = np.array(d["xrsa_flux"])
+            xrsb = np.array(d["xrsb_flux"])
+            times = parse_time("2000-01-01 12:00:00") + TimeDelta(d["time"], format="sec")
+
+        else:
+            raise ValueError("Don't know how to parse this file")
+
+        d.close()
+
+        data = DataFrame({"xrsa": xrsa, "xrsb": xrsb}, index=times.datetime)
+        data.replace(-9999, np.nan)
+        units = OrderedDict([("xrsa", u.W/u.m**2),
+                             ("xrsb", u.W/u.m**2)])
+
+        return data, header, units
+
     @classmethod
     def is_datasource_for(cls, **kwargs):
         """
         Determines if header corresponds to a GOES lightcurve
         `~sunpy.timeseries.TimeSeries`.
         """
-        if 'source' in kwargs.keys():
-            if kwargs.get('source', ''):
-                return kwargs.get('source', '').lower().startswith(cls._source)
-        if 'meta' in kwargs.keys():
-            return kwargs['meta'].get('TELESCOP', '').startswith('GOES')
+        if "source" in kwargs.keys():
+            if kwargs.get("source", ""):
+                return kwargs.get("source", "").lower().startswith(cls._source)
+        if "meta" in kwargs.keys():
+            return kwargs["meta"].get("TELESCOP", "").startswith("GOES")
+
+        if "filepath" in kwargs.keys():
+            return ("xrs" and ".nc" in kwargs.get("filepath", ""))
```

Comment:
A lot of my comments above about file type detection also apply here. It would probably be wise for this check to do at least the following in order:

* Check the file extension
* If file extension is unknown, open the file and see if it looks like a netcdf4 file.
* If the file looks like a netcdf4 file (based on file extension or opening it) see if it looks like a GOES file

return False if the last one isn't true, or you never make it to the last one. A lot of this logic could also be reused for the other places in this PR where we need to check type.